### PR TITLE
dev/core#5422 Fix no full message showing up for logged in registered users

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#5422 Fix no full message showing up for logged in registered users

Before
----------------------------------------
A logged in user who returns to the event info page will see the event full message - regardless of whether it is full

After
----------------------------------------
The logged in user should not see the event full message - although if the url does not have `&noFullMsg=true` in it they should see 

image

Technical Details
----------------------------------------
This appears to have regressed around the end of last year/ beginning of this - recent enough to target the rc but I don't think a port to stable is required right before the rc goes stable. However, as it was right before the rc is cut I added a lot of code comments point towards my thoughts about the 'right  fix' (stop overloading the `noFullMsg` url parameter for the `cid=0` / register someone else scenario

Comments
----------------------------------------

### CODE REVIEW

File: Civi/Test/PageWrapper.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  /**
+   * @param string $class
+   * @param array $urlParameters
+   */
+  public function __construct(string $class, array $urlParameters = []) {
+    $_GET = $urlParameters;
+    $this->page = new $class();
+    $_SERVER['REQUEST_METHOD'] = 'GET';
+    $_REQUEST = array_merge($_REQUEST, $urlParameters);
```
Comment (totten) [inline Civi/Test/PageWrapper.php:36]: @eileenmcnaughton  Have you seen `LocalHttpTest`? It's basically the same idea -- but more developed. There are bunch of usage-examples in [OembedTest](https://github.com/civicrm/civicrm-core/blob/5.77/ext/oembed/tests/phpunit/CRM/Oembed/Page/OembedTest.php#L26) (*search for `$this->client`*) and a few other places (like [testHttpUnsubscribe](https://github.com/civicrm/civicrm-core/blob/5f254d8b28d57d404b489004133f8ff16ac1fa25/tests/phpunit/CRM/Mailing/MailingSystemTestBase.php#L100)).

File: CRM/Event/Page/EventInfo.php [hunk 2]
```diff
@@ -271,17 +268,25 @@ public function run() {
...
-    ];
-
-    if (($this->isEventFull() && $noFullMsg === 'false') || CRM_Event_BAO_Event::checkRegistration($params)) {
-      $statusMessage = $this->getEventValue('event_full_text');
-      if (CRM_Event_BAO_Event::checkRegistration($params)) {
-        if ($noFullMsg == 'false') {
+    $isAlreadyRegistered = $this->isAlreadyRegistered();
+    // noFullMsg was originally passed in to suppress the message about the event being full. The intent
+    // was originally such that when you were sending the user back to the info page after registering
+    // they would not be told it was full. Along the way it got overloaded to encompass
```
Comment (totten) [inline CRM/Event/Page/EventInfo.php:274]: Aaaah. :bulb:  I was definitely scratching my head to understand that parameter. Comment helps a lot.

**Comment by totten** [comment]
`r-run`: Messages make more sense for the cases that I tried

Updated conditionals seem proportionate.

I'll take a gander at swapping the test helper.

**Comment by eileenmcnaughton** [comment]
@totten can we merge this & consider the test helper once the release is out - it's inline with existing stuff in similar code. I'm OK but nervous about your approach because I've had bad experiences running some of the tests you are pointing to so master is better IMHO

**Comment by totten** [comment]
I want to merge because it's a regression-fix attached to RC that's about to go out -- and the main patch seems fine.

I'm apprehensive about whether we will actually get around to cleaning up the test situation. But I hope we do.

**Pull request merged**